services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/controller/SegmentController.java

```diff
@@ -75,14 +81,18 @@ public SegmentWebResponseJson createSegment(@PathVariable String projectid, @Req
 
             // verify network type
             String networkType = segmentWebRequestObject.getNetworkType();
-            if (networkType.equals(NetworkTypeEnum.VXLAN)) {
-
-            } else if (networkType.equals(NetworkTypeEnum.VLAN)) {
-
-            }else if (networkType.equals(NetworkTypeEnum.GRE)) {
-
+            Long key = null;
+            if (NetworkTypeEnum.VXLAN.getNetworkType().equals(networkType)) {
+                key = segmentService.addVxlanEntity(segmentWebRequestObject.getId());
+            } else if (NetworkTypeEnum.VLAN.getNetworkType().equals(networkType)) {
+                key = segmentService.addVlanEntity(segmentWebRequestObject.getId(), networkTypeId, networkType);
```

Comment:
Do we need the parameter of networkType in the addVlanEntity method?

Reply:
Yes